frontend/packages/shared/src/components/GiteaHeader/VersionControlButtons/components/ShareChangesPopover/CommitAndPushContent/FileChangesInfoModal/FilePath/FilePathUtils.test.ts

```diff
@@ -0,0 +1,46 @@
+import { convertPureGitDiffToUserFriendlyDiff } from 'app-shared/components/GiteaHeader/VersionControlButtons/components/ShareChangesPopover/CommitAndPushContent/FileChangesInfoModal/FilePath/FilePathUtils';
+
+const diffStringMock = `diff --git a/fileName.json b/fileName.json
+index 0909a03..527e226 100644
+--- a/fileName.json
++++ b/fileName.json
+@@ -2,6 +2,30 @@
+- old line
++ new line
+unchanged line
+\ No newline at end of file`;
+
+describe('FilePathUtils', () => {
+  describe('convertPureGitDiffToUserFriendlyDiff', () => {
+    it('removes all metadata including line with double @ in the diff string', () => {
+      const userFriendlyDiff: string[] = convertPureGitDiffToUserFriendlyDiff(diffStringMock);
+
+      expect(userFriendlyDiff).toEqual(['- old line', '+ new line', 'unchanged line']);
+    });
+
+    it('removes last metadata line if it exists', () => {
+      const userFriendlyDiff: string[] = convertPureGitDiffToUserFriendlyDiff(diffStringMock);
+
+      expect(userFriendlyDiff).not.toContain('No newline at end of file');
+    });
+
+    it('returns last line if it is not metadata', () => {
+      diffStringMock.replace(' No newline at end of file', '');
+      const userFriendlyDiff: string[] = convertPureGitDiffToUserFriendlyDiff(diffStringMock);
+
+      expect(userFriendlyDiff).toEqual(['- old line', '+ new line', 'unchanged line']);
+    });
```

Comment:
Begge disse testene er vel allerede dekket av den første?

Reply:
Hvertfall den første! Men jeg tenkte den andre var grei å ha bare for å sjekke at ting ikke krasjer dersom diffen ikke skulle inneholdt den siste linjen

Reply:
Ah, jeg overså denne linjen:
`diffStringMock.replace(' No newline at end of file', '');`

Det er faktisk flere problemer med denne fremgangsmåten:
- `replace`-funksjonen endrer ikke tekststengen, den returnerer en ny. Så `diffStringMock` har fortsatt den originale teksten.
- Hvis `replace` _hadde_ endret tekststrengen, hadde det heller ikke vært helt heldig. Testene skal ikke være avhengige av å kjøre i en gitt rekkefølge, og derfor bør vi unngå å manipulere globale variabler.

Her er to forslag til løsning:
- Sett resultatet av `replace`-funksjonen inn i en egen variabel og kjør testen på denne.
- Definer hele tekststrengen på nytt slik den skal se ut for denne testen. Da blir det helt tydelig hva som testes. Vi får duplisert kode, men det er greit siden det brukes i tester.